tests/unit/blocking_connection_tests.py

```diff
@@ -205,6 +210,46 @@ def test_sleep(self, select_connection_class_mock):
                 spec_set=blocking_connection.BlockingConnection._flush_output):
             connection.sleep(0.00001)
 
+    def test_client_properties_default(self):
+        expectation = {
+            'product': connection.PRODUCT,
+            'platform': 'Python %s' % platform.python_version(),
+            'capabilities': {
+                'authentication_failure_close': True,
+                'basic.nack': True,
+                'connection.blocked': True,
+                'consumer_cancel_notify': True,
+                'publisher_confirms': True
+            },
+            'information': 'See http://pika.rtfd.org',
+            'version': pika.__version__
+        }
+        with mock.patch('pika.connection.Connection.connect'):
+            with self.assertRaises(exceptions.ConnectionClosed):
+                conn = blocking_connection.BlockingConnection()
+                self.assertDictEqual(conn._client_properties, expectation)
```

Comment:
I think that it might suffice only to test that the expected `client_properties` value was passed to the next layer (SelectConnection constructor). With unit tests, it might be better to isolate testing to the given layer/function. For this, as in `test_sleep`, simply patch `SelectConnection` and `_process_io_for_connection_setup`, then call `blocking_connection.BlockingConnection()`, and finally use `select_connection_class_mock.assert_called_once_with()` with the appropriate args to verify that the expected `client_properites` value was passed to SelectConnection (it would suffice to use `mock.ANY` for all args except `client_properties`) 

```
@patch.object(blocking_connection, 'SelectConnection',
                   spec_set=SelectConnectionTemplate)
```